modules/apps/transfer/keeper/relay.go

```diff
@@ -214,30 +193,24 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 		// NOTE: We use SourcePort and SourceChannel here, because the counterparty
 		// chain would have prefixed with DestPort and DestChannel when originally
 		// receiving this coin as seen in the "sender chain is the source" condition.
-		if types.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), fullDenomPath) {
+		if token.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel()) {
 			// sender chain is not the source, unescrow tokens
 
 			// remove prefix added by sender chain
-			voucherPrefix := types.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
-			unprefixedDenom := fullDenomPath[len(voucherPrefix):]
-
-			// coin denomination used in sending from the escrow address
-			denom := unprefixedDenom
+			token.Trace = token.Trace[1:]
 
-			// The denomination used to send the coins is either the native denom or the hash of the path
-			// if the denomination is not native.
-			denomTrace := types.ParseDenomTrace(unprefixedDenom)
-			if !denomTrace.IsNativeDenom() {
-				denom = denomTrace.IBCDenom()
+			// obtain the sdk coin for bank sends
+			coin, err := token.Coin()
```

Comment:
I think I'll likely remove this function since the amount is parsed above in this file